prowler/providers/aws/services/elasticache/elasticache_service.py

```diff
@@ -97,30 +122,24 @@ def __list_tags_for_resource__(self):
                     logger.error(
                         f"{regional_client.region} -- {error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
                     )
-        except Exception as error:
-            logger.error(
-                f"{regional_client.region} -- {error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
-            )
-
-    def __describe_replication_groups__(self, regional_client):
-        logger.info("Elasticache - Describing Replication Groups...")
-        try:
-            for repl_group in regional_client.describe_replication_groups()[
-                "ReplicationGroups"
-            ]:
-                replication_arn = repl_group["ARN"]
-                if not self.audit_resources or (
-                    is_resource_filtered(replication_arn, self.audit_resources)
-                ):
-                    self.replication_groups[replication_arn] = ReplicationGroup(
-                        id=repl_group["ReplicationGroupId"],
-                        arn=replication_arn,
-                        region=regional_client.region,
-                        status=repl_group["Status"],
-                        snapshot_retention=repl_group.get("SnapshotRetentionLimit", 0),
-                        encrypted=repl_group["AtRestEncryptionEnabled"],
-                        transit_encryption=repl_group["TransitEncryptionEnabled"],
-                        multi_az=repl_group["MultiAZ"],
+            for repl_group in self.replication_groups.values():
+                try:
+                    regional_client = self.regional_clients[repl_group.region]
+                    repl_group.tags = regional_client.list_tags_for_resource(
+                        ResourceName=repl_group.arn
+                    )["TagList"]
+                except ClientError as error:
+                    if error.response["Error"]["Code"] == "ReplicationGroupNotFound":
```

Comment:
Can you use here `ElastiCache.Client.exceptions.ReplicationGroupNotFoundFault` instead of doing the `if`? We need to start using that as we do here https://github.com/prowler-cloud/prowler/blob/169d1686d2620e64cff90bcaf0d4f5517d93473e/prowler/providers/aws/services/iam/iam_service.py#L284

Reply:
Love this!!